fix: Handle abbreviated name under self-managed properly

dawsontoth opened the pull request:
There were inconsistencies based on what you’d type, and when you’d switch, that could prevent Zod from passing validation. Or we’d accidentally pass an empty string to the server, causing a validation error there. This will avoid those odd cases, and make sure we pass along the abbreviated name when appropriate.

dawsontoth committed:
fix: Handle abbreviated name under self-managed properly

There were inconsistencies based on what you’d type, and when you’d switch, that could prevent Zod from passing validation. Or we’d accidentally pass an empty string to the server, causing a validation error there.

cap10morgan commented on src/features/clusters/upsert/index.tsx:
```diff
@@ -112,7 +112,7 @@ export function UpsertCluster() {
 			autoRenew: cluster?.plans?.[0]?.autoRenew ?? true,
 			systemName: cluster?.name ?? '',
-			abbreviatedName: cluster?.abbreviatedName ?? '',
+			abbreviatedName: cluster?.abbreviatedName || '',
```
Might benefit from a comment as to why this needs `||` when the other similar lines around it all use `??`?

dawsontoth replied:
Actually, with the solution I came up with around this, we can go back to the nullish coalescing operator!

cap10morgan replied:
> Actually, with the solution I came up with around this, we can go back to the nullish coalescing operator!

Even better! 👍🏻

cap10morgan reviewed (approved):
One small suggestion but looks good overall

dawsontoth committed:
chore: Go back to nullish coalescing

We didn’t need this line change, and this is more consistent with the code around it.
